pyscf/dft/rks.py

```diff
@@ -260,6 +260,46 @@ def define_xc_(ks, description, xctype='LDA', hyb=0, rsh=(0,0,0)):
     ks._numint = libxc.define_xc_(ks._numint, description, xctype, hyb, rsh)
     return ks
 
+# supported dispersion corrections
+DISP_VERSIONS = ['d3bj', 'd3zero', 'd3bjm', 'd3zerom', 'd3op', 'd4']
+
+def dft_method_parser(dft_method):
+    ''' conventional dft method ->
+    ( internal xc,
+      disable nlc,
+      dispersion version
+      with 3-body dispersion or not
+      '''
+    method_lower = dft_method.lower()
```

Comment:
Please check if xc is a string and return early. The input xc can be a tuple of strings or other data types.